src/daft-table/src/ops/joins/mod.rs

```diff
@@ -103,14 +108,96 @@ pub fn infer_join_schema(
     Schema::new(join_fields)
 }
 
+fn add_non_join_key_columns(
+    left: &Table,
+    right: &Table,
+    lidx: Series,
```

Comment:
why not take these as `&Series`

Reply:
I move `lidx` so that I can drop it early to save memory

Reply:
ah makes sense